stan/math/prim/mat/meta/vector_seq_view.hpp

```diff
@@ -0,0 +1,85 @@
+#ifndef STAN_MATH_PRIM_MAT_META_VECTOR_SEQ_VIEW_HPP
+#define STAN_MATH_PRIM_MAT_META_VECTOR_SEQ_VIEW_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <vector>
+
+namespace stan {
+  using std::vector;
+  using Eigen::Matrix;
+  using Eigen::Dynamic;
+
+  /**
+   * Provides a low-cost wrapper for situations where you either need an Eigen
```

Comment:
How is this?
   * This class provides a low-cost wrapper for situations where you either need
   * an Eigen Vector or RowVector or a std::vector of them and you want to be
   * agnostic between those two options. This is similar to scalar_seq_view but
   * instead of being a sequence-like view over a scalar or seq of scalars, it's
   * a sequence-like view over a Vector or seq of Vectors. Notably this version
   * only allows std::vectors as the container type, since we would have
   * difficulty figuring out which contained type was the container otherwise.
   *
   * @tparam T the underlying scalar type inside the underlying Vector